src/app.ts

```diff
@@ -23,61 +24,56 @@ import { EthereumBlockchainService } from './services/blockchain/ethereum/ethere
 import { ServiceMetrics as Metrics } from './service-metrics.js'
 import { version } from './version.js'
 import { cleanupConfigForLogging, normalizeConfig } from './normalize-config.util.js'
+import type { Injector } from 'typed-inject'
+import type { EventProducerService } from './services/event-producer/event-producer-service.js'
+
+type DependenciesContext = {
+  config: Config
+  dbConnection: Knex
+}
+
+type ProvidedContext = {
+  anchorService: AnchorService
+  requestRepository: RequestRepository
+  anchorRepository: AnchorRepository
+  transactionRepository: TransactionRepository
+  blockchainService: BlockchainService
+  eventProducerService: EventProducerService
+  ceramicService: CeramicService
+  ipfsService: IpfsService
+  schedulerService: SchedulerService
+} & DependenciesContext
 
 /**
  * Ceramic Anchor Service application
  */
 export class CeramicAnchorApp {
   private _schedulerService: SchedulerService
   private _server: CeramicAnchorServer
+  readonly container: Injector<ProvidedContext>
+  private readonly config: Config
 
-  constructor(
-    private readonly container: DependencyContainer,
-    private readonly config: Config,
-    dbConnection: Knex
-  ) {
-    normalizeConfig(config)
+  constructor(container: Injector<DependenciesContext>) {
+    this.config = container.resolve('config')
+    normalizeConfig(this.config)
 
     // TODO: Selectively register only the global singletons needed based on the config
 
-    // register config
-    container.register('config', {
-      useFactory: instanceCachingFactory<Config>((c) => config),
-    })
-
-    // register database connection
-    container.register('dbConnection', {
-      useFactory: instanceCachingFactory<Knex>((c) => dbConnection),
-    })
-
-    // register repositories
-    container.registerSingleton('requestRepository', RequestRepository)
-    container.registerSingleton('anchorRepository', AnchorRepository)
-    container.registerSingleton('transactionRepository', TransactionRepository)
-
-    // register services
-    container.register('blockchainService', {
-      useFactory: instanceCachingFactory<EthereumBlockchainService>((c) =>
-        EthereumBlockchainService.make(config)
-      ),
-    })
-    container.registerSingleton('eventProducerService', HTTPEventProducerService)
-    container.registerSingleton('anchorService', AnchorService)
-    container.registerSingleton('ceramicService', CeramicServiceImpl)
-    container.registerSingleton('ipfsService', IpfsServiceImpl)
-    container.registerSingleton('schedulerService', SchedulerService)
-
-    // register controllers
-    container.registerSingleton('healthcheckController', HealthcheckController)
```

Comment:
my comment about the healthcheck controller is due to this change.  We used to always register this controller, now we only register it if we call `_startServer`